core/dbt/config/project.py

```diff
@@ -645,13 +646,26 @@ def from_project_root(
     def hashed_name(self):
         return hashlib.md5(self.project_name.encode('utf-8')).hexdigest()
 
-    def get_selector(self, name: str) -> SelectionSpec:
+    def get_selector(self, name: str) -> Union[SelectionSpec, bool]:
         if name not in self.selectors:
             raise RuntimeException(
                 f'Could not find selector named {name}, expected one of '
                 f'{list(self.selectors)}'
             )
-        return self.selectors[name]
+        return self.selectors[name]["definition"]
+
+    def get_default_selector(self) -> Union[SelectionSpec, bool, None]:
+        """This function fetch the default selector to use on `dbt run` (if any)
+        :return: either a selector if default is set or None
+        :rtype: Union[SelectionSpec, None]
+        """
+        for selector_name, selector in self.selectors.items():
+            if selector["default"] is True:
+                name = selector_name
+                print_timestamped_line(f'Using default selector {name}')
```

Comment:
Really happy to have logging that will help avoid user confusion. Two quick thoughts:
- [cosmetic] I think this should just be `logger.info` rather than `print_timestamped_line`. It feels more like the `Found 5 models, ...` line (which doesn't include a timestamp), versus the actual execution steps that print with timestamps below it.
- [functional] We don't want to log this line every time the user has a default selector _defined_, only if it's being _used_ (i.e. no CLI `--select` or `--exclude` args). So I think this logging needs to go in the `get_selection_spec` task method instead, in the second conditional branch.

Since that's one more line of code to add in three places, I'm thinking: All of these tasks now use the same exact definition of `get_selection_spec` (thanks to #3554). What do you think of just updating the `GraphRunnableTask` method, so that instead of implementing as an `abstractmethod` it actually defines the logic that `compile`, `list`, and `freshness` will all use? Then we can delete the duplicated code from `compile`, `list`, and `freshness`, since they'll just inherit from `GraphRunnableTask` (just as `run`, `test`, etc inherit from `compile`).

https://github.com/dbt-labs/dbt/blob/3effade26603a38fa71b10990a82e3318faad7b8/core/dbt/task/runnable.py#L106-L110

Reply:
Do you know why list.py had `spec = parse_difference(self.selector, ...` in `def get_selection_spec` while freshness and compile have `spec = parse_difference(self.args.select` ?

Reply:
@TeddyCr Ah, it's just because `list` accepts both `models` and `select`. So options are:
- reimplement `get_selection_spec` for `list` alone
- implement `selector` as a property on `GraphRunnableTask`, and override just for `ListTask`

I'm leaning toward the latter